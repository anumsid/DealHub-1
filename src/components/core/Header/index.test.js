import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { Header } from './index'

describe('Header Component', () => {
  let wrapper
  let logout
  beforeEach(() => {
    logout = stub()
    const props = {
      authToken: 'xxx',
      location: { pathname: '' },
      logout: logout,
    }
    wrapper = shallow(<Header {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('logs out user', () => {
    wrapper.find('Link[to="#"]').simulate('click')
    expect(logout.calledOnce).toBe(true)
  })
})
