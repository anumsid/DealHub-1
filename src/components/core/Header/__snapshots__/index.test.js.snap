// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Matches snapshot 1`] = `
<div
  className="fixed-top"
>
  <Connect(Component) />
  <Connect(SearchDeals) />
  <div
    className="navbar-collapse"
    id="navbarNavAltMarkup"
  >
    <div
      className="navItems"
    >
      <Link
        className="linkItem text-light"
        replace={false}
        to="/"
      >
        Home
      </Link>
      <Link
        className="linkItem text-light"
        onClick={[Function]}
        replace={false}
        to="#"
      >
        Logout
      </Link>
    </div>
  </div>
  <header
    className="App-header align-content-center"
  >
    <h1
      className="App-title"
    >
      Welcome to DealHub
    </h1>
  </header>
</div>
`;
